mlir/include/mlir/IR/OperationSupport.h

```diff
@@ -1039,6 +1039,8 @@ struct OperationState {
 
   /// Add an attribute with the specified name.
   void addAttribute(StringAttr name, Attribute attr) {
```

Comment:
Nit: Maybe consider to extend the, albeit limited documentation of these functions to incorporate these pre-conditions.